Add Facebook Folly port

KindDragon opened the pull request.

KindDragon committed:
[folly] Initial commit for Facebook Folly library

ras0219-msft committed:
[vcpkg-internalCI] Add folly:x64-windows to internal CI builds

ras0219-msft commented:
Thanks so much for making this happen 🎈! Really excited to see Folly on Windows (and in Vcpkg ;))!

I'm testing it in our CI system now, I'll merge once it completes.

ras0219-msft committed:
[vcpkg][folly] Fix post lint checks for folly.

ras0219-msft commented:
I needed to fix two things:
- During the build, the fingerprint generator required DLLs from the installed folder. In your WIP PR, I remembered that you copied it -- this time, I have just added the installed bin folder to the path.
- Vcpkg was erroring on the `debug/lib/folly.lib` -- I suspect our COFF reader has a bug that doesn't properly pick up the library's architectures. I've added a special allowance for "zero architecture libs" which we can hopefully close in the future once we diagnose and fix the COFF reader bug.

KindDragon commented:
> During the build, the fingerprint generator required DLLs from the installed folder. In your WIP PR, I remembered that you copied them -- this time, I have just added the installed bin folder to the path.

It's only required with dynamic linkage, because of that I removed them.
